.circleci/config.yml

```diff
@@ -172,6 +172,42 @@ commands:
       - store_test_results:
           path: .
 
+  publish:
+    steps:
+      - attach_workspace:
+          at: .
+
+      - run:
+          name: Remove unnecessary files
+          command: rm -rf emsdk
+
+      - run:
+          name: Update version if not on git tag
+          command: |
+            if ! git describe --exact-match HEAD 1>/dev/null 2>/dev/null
+            then
+              git describe --exact-match HEAD 1>/dev/null 2>/dev/null && exit
+              CURRENT_VERSION="$(node -p "require('./package.json').version")"
+              GIT_COMMIT_SHA="$(git rev-parse --short HEAD)"
+              npm version "$CURRENT_VERSION-$GIT_COMMIT_SHA"
+            fi
+
+      - run:
+          name: Login to npm
+          command: npm set //registry.npmjs.org/:_authToken=$NPM_TOKEN
+
+      - run:
+          name: Publish to npm
+          command: |
+            if git describe --exact-match HEAD 1>/dev/null 2>/dev/null
+            then
+              # Publish to latest when a git tag is pushed.
+              npm publish --tag latest
+            else
+              # Publish to beta when not run on a git tag.
+              npm publish --tag beta
```

Comment:
This is currently timing out on CI because it requires an otp - https://app.circleci.com/pipelines/github/postmanlabs/postject/109/workflows/92460ff1-b326-4bab-a273-c51afbfb3790/jobs/760?invite=true#step-106-29:
```
...
This operation requires a one-time password.
Enter OTP: 

Too long with no output (exceeded 10m0s): context deadline exceeded
```
although I have enabled `Require two-factor authentication or automation tokens` in https://www.npmjs.com/package/postject/access.

FYI, `NPM_TOKEN` is equal to the auth token of my primary npm account (it's the same as the value stored in `~/.npmrc`).

Not sure what's going wrong here, some help would be appreciated!